judge/runpipe.cc

```diff
@@ -257,7 +257,7 @@ void pump_pipes(int *fd_out, int *fd_in, int from_val)
 	FD_SET(*fd_out, &readfds);
 
 	tv.tv_sec = 0;
-	tv.tv_usec = 1000; /* FIXME: this is just in order to not block */
+	tv.tv_usec = 20; /* FIXME: this is just in order to not block */
```

Comment:
A lower timeout here means processing data with less latency, but also more polling cycles spent by the `runpipe` program, possibly increasing total wallclock of the whole judging run at a fixed amount of CPU time spent by the submission program. It would be nice to reason about or test this in a bit more detail, but given that this was used without problems at BAPC, I'm ok with this choice.